### PR TITLE
chall09 isoria-g

### PR DESCRIPTION
**Pull request by nsoria042**
The first thing is to analyze the input string and verify that it is a valid number to work with and I use ATOI (controlling characters before and after). Then I check the "Invalid input".
The number I work with ATOI. The first thing is to analyze the input string and verify that it is a valid number to work with and then I use ATOI (controlling characters before and after). Then I check the "Invalid input".
I work the number dividing by the units in seconds and I build the string with strjoin and ITOA.

### CODE REVIEW

**Comment by nsoria042** [comment]
The first thing is to analyze the input string and verify that it is a valid number to work with and I use ATOI (controlling characters before and after). Then I check the "Invalid input".
The number I work with ATOI. The first thing is to analyze the input string and verify that it is a valid number to work with and then I use ATOI (controlling characters before and after). Then I check the "Invalid input".
I work the number dividing by the units in seconds and I build the string with strjoin and ITOA.

File: chall09/isoria-g.c [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+size_t      ft_strlen(const char *s)
+{
+    size_t  a;
+
+    a = 0;
+    while (s[a] != '\0')
+        a++;
+    return (a);
+}
```
Comment (clafoutis42) [inline chall09/isoria-g.c:28]: Why not use standard library's `strlen` ?

File: chall09/isoria-g.c [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    str1 = (char *)t;
+    while (a < (l1 + l2))
+    {
+        if (a < l1)
+            str1[a] = s1[a];
+        else
+            str1[a] = s2[a - l1];
+        a++;
+    }
+    str1[a] = '\0';
```
Comment (clafoutis42) [inline chall09/isoria-g.c:54]: You could have used `strcpy` and `strcat` to do this ;)

File: chall09/isoria-g.c [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    unsigned int    l1;
+    unsigned int    l2;
+    char            *t;
+    char            *str1;
+
+    a = 0;
+    if (s1 == NULL || s2 == NULL)
+        return (NULL);
+    l1 = (unsigned int)ft_strlen((char *)s1);
+    l2 = (unsigned int)ft_strlen((char *)s2);
```
Comment (clafoutis42) [inline chall09/isoria-g.c:42]: Why are you casting `s1` and `s2` ? They already are of type `char *`

File: chall09/isoria-g.c [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+static unsigned long long   ft_value(unsigned long long n)
+{
+    unsigned long long      b;
+
+    if (n < 0)
+        b = n * (-1);
+    else
+        b = n;
+    return (b);
+}
```
Comment (clafoutis42) [inline chall09/isoria-g.c:67]: Unsigned variables can never be negative

File: chall09/isoria-g.c [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+char                    *ft_format_duration(char *seconds)
+{
+    unsigned long long  nb;
+    unsigned long long  yy;
+    unsigned long long  dd;
+    unsigned long long  hh;
+    unsigned long long  mm;
+    unsigned long long  ss;
+    char *str = "";
```
Comment (clafoutis42) [inline chall09/isoria-g.c:188]: It is better to initiate it with `NULL`

File: chall09/isoria-g.c [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+    if (seconds == NULL)
+        return("\"Invalid input.\"");
+    nb = ft_atoi(seconds);
+    if (nb == -1)
+        return("\"Invalid input.\"");
+    if (nb == 0)
+        return("\"now\"");
```
Comment (clafoutis42) [inline chall09/isoria-g.c:196]: Why did you return this values with a quote?

File: chall09/isoria-g.c [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+    yy = nb / 31536000;
+    dd = (nb % 31536000) / 86400;
+    hh = ((nb % 31536000) % 86400) / 3600;
+    mm = (((nb % 31536000) % 86400) % 3600) / 60;
+    ss = (((nb % 31536000) % 86400) % 3600) % 60;
```
Comment (clafoutis42) [inline chall09/isoria-g.c:201]: Using macros to define those numbers by what they represent would have helped readability

File: chall09/isoria-g.c [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+        if (yy == 1)
+        {
+            str = ft_strjoin(str, ft_itoa(yy));
+            str = ft_strjoin(str, " year");
+        }
+        else
+        {
+            str = ft_strjoin(str, ft_itoa(yy));
+            str = ft_strjoin(str, " years");
+        }
```
Comment (clafoutis42) [inline chall09/isoria-g.c:214]: This seems overly repetitive. You should have made a function for this.  
Even better you could have used functions like `sprintf` or other variations of it to format your string easily

File: chall09/isoria-g.c [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+            str = ft_strjoin(str, ft_itoa(ss));
+            str = ft_strjoin(str, " second");
+        }
+        else
+        {
+            str = ft_strjoin(str, ft_itoa(ss));
+            str = ft_strjoin(str, " seconds");
+        }
+    }
+    str = ft_strjoin(str, "\"");
```
Comment (clafoutis42) [inline chall09/isoria-g.c:333]: Why are you quoting the result ? :/

**Review by clafoutis42 — COMMENTED** [review]
Your code seems to work, but you are overly repetitive and you added quotes for a reason I don't get.  
Unfortunately this won't work, but nice try!

**Comment by clafoutis42** [comment]
I just noticed your commit was initially late, I'm gonna have to revert this PR, sorry.

**Pull request merged**